lib/v2/screens/profile_screens/profile/interactor/mappers/upgrade_citizenship_result_mapper.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:seeds/v2/domain-shared/page_command.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/page_commands.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/profile_state.dart';
+
+class UpgradeCitizenshipResultMapper extends StateMapper {
+  ProfileState mapResultToState(ProfileState currentState, Result result, bool isResident) {
+    String newCitizenshipStatus;
+
+    if (result.isError) {
+      // citizenship upgrade error, show snackbar fail
+      print('Error upgrading citizenship status');
+      return currentState.copyWith(
+        pageState: PageState.success,
+        pageCommand: ShowErrorMessage('Fail to upgrade citizenship status.'),
+      );
+    } else {
+      // Show citizenship upgrade success
+      if (isResident) {
+        newCitizenshipStatus = "Citizen";
+      } else {
+        newCitizenshipStatus = "Resident";
+      }
```

Comment:
I prefer one line logic.

`final String newCitizenshipStatus = isCitizen? "Citizen": "Resident";`

Reply:
Done, Thanks for the suggestions